dggbot/bot.py

```diff
@@ -106,8 +108,11 @@ def _post_message(self, msg: Message):
             self.on_command(msg)
 
     def send(self, msg: str):
-        if self._last_send_time is not None and self._send_cooldown:
-            if (time.time() - self._last_send_time) < self._send_cooldown:
-                return
+        if self._last_msg is not None:
+            if self._send_cooldown:
+                if time.time() - self._last_msg[1] < self._send_cooldown:
+                    return
+            if self._avoid_dupe and msg == self._last_msg[0]:
+                msg += " ."
         super().send(msg)
-        self._last_send_time = time.time()
+        self._last_msg = (msg, time.time())
```

Comment:
An issue that could arise is when a msg isn't sent for whatever reason (either duplicate for whatever reason that wasn't detected, throttled, sub-only chat, when muted), a way to tell if a msg was properly sent if possible would be good but probably outside the scope of this and would require a lot more overhaul. Might do that at some point in the future, this is more so just a note for the future.